blockchain/common_test.go

```diff
@@ -270,6 +270,33 @@ func findDeploymentChoice(deployment *chaincfg.ConsensusDeployment, choiceID str
 	return nil, fmt.Errorf("unable to find vote choice for id %q", choiceID)
 }
 
+// Make the linter happy.  This should be removed once the function is actually
+// used.
+var _ = removeDeployment
+
+// removeDeployment modifies the passed parameters to remove all deployments for
+// the provided vote ID.  An error is returned when not found.
+func removeDeployment(params *chaincfg.Params, voteID string) error {
+	// Remove the deployment(s) for the passed vote ID.
+	var found bool
```

Comment:
I'd use `findDeployment` and delete the deployment that returns.

Reply:
That won't work because it only returns a pointer to it, not its index.

Reply:
It returns a uint32 that can be straight deleted.

Reply:
That is the deployment version, not the deployment ID.  You would remove all deployments for the entire agenda version using that value.

Reply:
```
        const tVoteID = chaincfg.VoteIDTreasury
        params = cloneParams(params)
        tVersion, deployment, err := findDeployment(params, tVoteID)
        if err != nil {
                t.Fatal(err)
        }
        removeDeploymentTimeConstraints(deployment)
        delete(params.Deployments, tVersion)
```

Reply:
As I said, that deletes the entire agenda version, including *all* deployments for that version, not the specific deployment.

Reply:
As a case in point, try doing that with `chaincfg.VoteIDLNSupport`.  You will end up removing both it and `chaincfg.VoteIDSDiffAlgorithm`, because they are both deployment version 4.

See: https://github.com/decred/dcrd/blob/98f0dd457b226f15734ef3d92893e6aea21f7766/chaincfg/mainnetparams.go#L150-L205